generate_engagement_analysis_files.py

```diff
@@ -0,0 +1,152 @@
+                        help="Directory to write engagement CSV files to")
+
+
+    args = parser.parse_args()
+
+    user = args.user
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    demog_map_json_input_dir = args.demog_map_json_input_dir
+    engagement_csv_output_dir = args.engagement_csv_output_dir
+
+    # Load the pipeline configuration file
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+
+    # Infer which RQA coding plans to use from the pipeline name.
+    # Starting from q4 pipeline since the pipeline was refactored into quarterly pipelines when the project had reached
+    # q4 reporting. Earlier episodes are covered in the full pipeline configurations
+    if pipeline_configuration.pipeline_name == "q4_pipeline":
+        log.info("Extracting Q4 pipeline data")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.Q4_RQA_CODING_PLANS
+    elif pipeline_configuration.pipeline_name == "q5_pipeline":
+        log.info("Extracting Q5 pipeline data")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.Q5_RQA_CODING_PLANS
+    elif pipeline_configuration.pipeline_name == "q6_pipeline":
+        log.info("Extracting Q6 pipeline data")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.Q6_RQA_CODING_PLANS
+    else:
+        assert pipeline_configuration.pipeline_name == "full_pipeline", "PipelineName must be either 'a quartely pipeline name' or 'full pipeline'"
+        log.info("Extracting full pipeline data")
+        PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.FULL_PIPELINE_RQA_CODING_PLANS
+
+    # Create a list of rqa_raw_fields
+    rqa_raw_fields = []
+    for plan in PipelineConfiguration.RQA_CODING_PLANS:
+        rqa_raw_fields.append(plan.raw_field)
+
+    log.info(f'Computing unique, lifetime_activations_per_show and per-show participants ...' )
+    engagement_map = {}  # of uid -> name of shows participated in and their demographics data.
+    total_participants_with_optins= set()  # total of participants who sent a message and also opted in.
+    total_activations_with_optins = []  # sum of total of every time consented participants interacts throughout a project.
+    activations_per_show_with_optins = OrderedDict()  # of rqa_raw_field -> sum of total of every time consented participants interact in an episode.
+    for rqa_raw_field in rqa_raw_fields:
+        with open(f'{demog_map_json_input_dir}/{rqa_raw_field}_demog_map.json') as f:
+            data = json.load(f)
+        log.info(f"Loaded {len(data)} {rqa_raw_field} uids ")
+
+        activations_per_show_with_optins[f"{rqa_raw_field}"] = len(data.keys())
+
+        for uid, demogs in data.items():
+            demog = data[uid]
```

Comment:
I find this a bit confusing.

demogs == data[uid], so makes more sense to use demogs here.
Then you would have demog = demogs...

How about just setting the demographics inline on L79 i.e. "demogs": demogs